src/main/scala/org/allenai/deep_qa/data/OmnibusDaDatasetReader.scala

```diff
@@ -26,31 +26,48 @@ class OmnibusDaDatasetReader extends DatasetReader[DirectAnswerInstance] {
       answerText = line(4)
     } yield (questionText, answerText)
     val instances = instanceTuples.map { case (questionText, answerText) => {
-      DirectAnswerInstance(questionText, Some(answerText))
+      DirectAnswerInstance(imperativeToInterrogative(questionText), Some(answerText))
     }}
     Dataset(instances)
   }
 
   def imperativeToInterrogative(inputImperative: String): String = {
     val imperativeSentences = Parser.stanford.splitSentences(inputImperative)
-    val imperatives = """identify|name|define|give|explain|describe|list"""
-    val interrogatives = """who|what|when|where|why|how"""
+    val interrogatives = Seq("who", "what", "when", "where", "why", "how")
+    val interrogativesStr = interrogatives.mkString("|")
+
+    val imperatives = Seq("identify", "name", "define", "give", "explain", "describe", "list")
+    val imperativesStr = imperatives.mkString("|")
 
     val imperativeModifierRemovedSentences = for {
       imperativeSentence <- imperativeSentences
 
       // If there is an imperative, remove all words that occur before
       // it (the first imperative) in the sentence. Thus, "Please briefly explain this." becomes
-      // "explain this."
-      imperativeModifiers = s"""(?i).+?(?=(${imperatives}) )""".r
-      imperativeModifierRemovedSentence = imperativeModifiers.replaceFirstIn(imperativeSentence, "")
+      // "explain this.". However, if the first word is an imperative or an interrogative word,
+      // do not remove anything.
+      imperativeModifiers = s"""(?i).+?(?=(${imperativesStr}) )""".r
+      imperativeModifierRemovedSentence = if (interrogatives.contains(
+        imperativeSentence.toLowerCase.split(" ").head) ||
+        imperatives.contains(
+          imperativeSentence.toLowerCase.split(" ").head)
```

Comment:
Also, I'd probably add a `firstWord = imperativeSentence.toLowerCase.split(" ").head` line, to make this more readable and simpler.